File: code_samples/typescript/oop_encapsulation.ts
/**
 * Oop: Encapsulation
 * AI/ML Training Sample
 */

interface IEncapsulation {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Encapsulation implements IEncapsulation {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Encapsulation();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Encapsulation, IEncapsulation };

File: code_samples/r/oop_encapsulation.r
# Oop: Encapsulation
# AI/ML Training Sample

Encapsulation <- setRefClass(
    "Encapsulation",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Encapsulation$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
